src/ProjectTemplates/Web.ProjectTemplates/content/Components-CSharp/.template.config/template.json

```diff
@@ -0,0 +1,287 @@
+      }
+    },
+    "kestrelHttpPortReplacer": {
+      "type": "generated",
+      "generator": "coalesce",
+      "parameters": {
+        "sourceVariableName": "kestrelHttpPort",
+        "fallbackVariableName": "kestrelHttpPortGenerated"
+      },
+      "replaces": "5000"
+    },
+    "kestrelHttpsPort": {
+      "type": "parameter",
+      "datatype": "integer",
+      "description": "Port number to use for the HTTPS endpoint in launchSettings.json. This option is only applicable when the parameter no-https is not used (no-https will be ignored if either IndividualAuth or OrganizationalAuth is used)."
+    },
+    "kestrelHttpsPortGenerated": {
+      "type": "generated",
+      "generator": "port",
+      "parameters": {
+        "low": 7000,
+        "high": 7300
+      }
+    },
+    "kestrelHttpsPortReplacer": {
+      "type": "generated",
+      "generator": "coalesce",
+      "parameters": {
+        "sourceVariableName": "kestrelHttpsPort",
+        "fallbackVariableName": "kestrelHttpsPortGenerated"
+      },
+      "replaces": "5001"
+    },
+    "iisHttpPort": {
+      "type": "parameter",
+      "datatype": "integer",
+      "description": "Port number to use for the IIS Express HTTP endpoint in launchSettings.json."
+    },
+    "iisHttpPortGenerated": {
+      "type": "generated",
+      "generator": "port"
+    },
+    "iisHttpPortReplacer": {
+      "type": "generated",
+      "generator": "coalesce",
+      "parameters": {
+        "sourceVariableName": "iisHttpPort",
+        "fallbackVariableName": "iisHttpPortGenerated"
+      },
+      "replaces": "8080"
+    },
+    "iisHttpsPort": {
+      "type": "parameter",
+      "datatype": "integer",
+      "description": "Port number to use for the IIS Express HTTPS endpoint in launchSettings.json. This option is only applicable when the parameter no-https is not used (no-https will be ignored if either IndividualAuth or OrganizationalAuth is used)."
+    },
+    "iisHttpsPortGenerated": {
+      "type": "generated",
+      "generator": "port",
+      "parameters": {
+        "low": 44300,
+        "high": 44399
+      }
+    },
+    "iisHttpsPortReplacer": {
+      "type": "generated",
+      "generator": "coalesce",
+      "parameters": {
+        "sourceVariableName": "iisHttpsPort",
+        "fallbackVariableName": "iisHttpsPortGenerated"
+      },
+      "replaces": "44300"
+    },
+    "UseClient": {
+      "type": "parameter",
+      "datatype": "bool",
+      "defaultValue": "false",
+      "displayName": "_Use interactive webassembly components",
+      "description": "If specified, configures the project to render components interactively on the browser using webassembly."
+    },
+    "UseServer": {
+      "type": "parameter",
+      "datatype": "bool",
+      "defaultValue": "false",
+      "displayName": "_Use interactive server components",
+      "description": "If specified, configures the project to render components interactively on the server."
```

Comment:
It feels weird to me that the default Blazor template would have no interactivity enabled 🤔. I get why it's done this way. But I want to make sure we've discussed all the options:

1. No interactivity by default. Counter doesn't work. You have to manually add support for interactivity, which is pretty easy for Blazor Server, but more complicated for Blazor WebAssembly (Add TFM, setup the WebAssembly host)
2. Server interactivity by default, no WebAssembly. You're already running on the server. But if you decide you don't want to use Server then you have the Blazor hub endpoint exposed that you don't really need, but maybe it's harmless?
3. Both Server & WebAssembly enabled by default. Everything works, including auto, which presumably we'd use by default. But the template is more complicated. You'd need a way to turn off WebAssembly in case you don't want it.

I think we should make the most common case the default. I lean toward the third option of having interactivity modes on by default, with the option to turn either of them off. Thoughts?

Reply:
I wouldn't read much into this. We can flip the default values at the end. I started here to make sure that `dotnet new blazor` always creates something that works.

I am ok flipping the default values later in the release (once we have all interactive modes configured). In the end, the options are "sticky", so the user will remember.

Also, these are checkboxes (separate options) but we could make them a choice (select menu)

Reply:
I'll settle this as; we'll flip the defaults later on as we see fit when we enable the appropriate functionality.

Reply:
In VS I get that these can be checkboxes that are enabled/disabled by default and sticky. But on the CLI if we change the defaults doesn't that mean we need a way to negate the options?

Reply:
No, the option is separate from the symbol, we will redefine the `UseWasm` symbol in terms of the `NoWasm` parameter. Or we will make the parameter a choice value.